src/aws-manager.js

```diff
@@ -285,7 +260,12 @@ class AwsManager {
             let workerType = this.parseKeyPairName(instance.KeyName).workerType;
             instance.Region = region;
             instance.WorkerType = workerType;
-            apiState.instances.push(instance);
+            // apiState for instance state name of 'running', 'pending'
+            if (_.includes(['running', 'pending'], instance.State.Name)) {
+              apiState.instances.push(instance);
+            } else if (_.includes(['shutting-down', 'terminated', 'stopping'], instance.State.Name)) {
+              deadState.instances.push(instance);
+            }
             ignoreSR.push(instance.SpotInstanceRequestId);
```

Comment:
hmm... are you not missing some return statements? I'm not following the logic I'll admit... I don't know what `ignoreSR` is good for..